extpsdk/src/main/python/vantiq/extpsdk/VantiqConnector.py

```diff
@@ -0,0 +1,759 @@
+                    else:
+                        await self._process_message(op, message)
+                elif _STATUS in message:
+                    # Then this is a http response message.  Accept that if it's OK.
+                    if message[_STATUS] >= 300:
+                        # Then we have some error condition.  Log it & move on
+                        _vlog.error('Received status message indicating a problem: {0}'.format(message))
+                else:
+                    _vlog.error('Malformed message received from server: %s', message)
+            return retval
+        finally:
+            old_future = self._is_connected_future
+            self._is_connected_future = asyncio.get_event_loop().create_future()
+            if old_future is not None and not old_future.done():
+                # Mark a cancelled state so that any waiters will find out
+                old_future.cancel()
+            self.connection = None
+
+    async def close(self):
+        """(Async) Close the connection to the Vantiq server."""
+        if self.connection is not None:
+            await self.connection.close()
+
+    async def _process_message(self, op: string, message: dict):
+        global _vlog
+        global _OP_PUBLISH
+        global _OP_QUERY
+
+        local = None
+        if op == _OP_PUBLISH:
+            local = self.publish_handler
+        elif op == _OP_QUERY:
+            local = self.query_handler
+        else:
+            _vlog.error('Unexpected operation: %s -- ignored', op)
+
+        if local is not None:
+            try:
+                await local(self._create_context(message),
+                            message[_OBJECT] if message is not None and _OBJECT in message else None)
+            except Exception as e:  # pylint: disable=broad-except
+                _vlog.error('Exception {0} thrown by {1} handler while processing message {2}'
+                            .format(op, type(e).__name__, message))
+                _vlog.error('Trace: %s', traceback.format_exc())
+        else:
+            _vlog.error('No handler found for operation %s', op)
+
+    async def send_query_error(self, ctx: dict, message: dict) -> None:
+        """(Async) Respond to a Vantiq query with an error
+
+        Parameters:
+            ctx : dict
+                The context for the query.  Contains the 'source_name' and 'response_address' entries.
+            message : dict
+                The error message to return.  The error message should contain the following entries:
+                    'messageCode' --  a string containing a short name for the error
+                    'messageTemplate'  -- a string describing the problem.  Parameters in this template are indexed
+                                        from 0, and indicated using {index}.
+                    'parameters' -- a list/array containing the parameters to be substituted in the template.
+
+        Examples:
+        ::
```

Comment:
This `::` makes the documentation processor treat things like the \`\`\` operator in Markdown, not formatting things.